clients/input_copy.go

```diff
@@ -33,14 +38,33 @@ type InputCopy struct {
 }
 
 // CopyInputToS3 copies the input video to our S3 transfer bucket and probes the file.
-func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVideoProbe video.InputVideo, signedURL string, osTransferURL *url.URL, err error) {
+func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL, encryptedKey string, vodEncryptPrivateKey string) (inputVideoProbe video.InputVideo, signedURL string, osTransferURL *url.URL, err error) {
+
+	// Create a temporary local file to write to
+	localSourceFile, err := os.CreateTemp(os.TempDir(), LocalSourceFilePattern)
+	if err != nil {
+		glog.Errorf("failed to create local file on vod upload: %s", err)
+		return
+	}
+	defer localSourceFile.Close()
+	defer os.Remove(localSourceFile.Name()) // Clean up the file as soon as we're done segmenting
```

Comment:
I think that ideally we would refactor the file downloading to be in a single place instead, on the outermost layer (runner) and make sure that we create&delete a single local file for the job. This would help reduce all the repeated downloading of the file that seems to happen today inside catalyst (only in this context: download once for s3 copy, once for probe, once for ffmpeg segmenting).

If that is too complex of a refactor to make now, I thought that this current implementation is _already_ an improvement over the previous state: instead of downloading the file once for copying to s3 and once for probing it, we download only once and use it for both (with the added decryption). So it's already an improvement. 

I think this setup is also better than another download&upload just to decrypt the file, followed by other downloads to copy to s3, probe, segment etc. So the fact that we are not re-using the file from here for segmenting is a separate issue to be optimized later, but it is not a regression related to the previous state (it seems more like an improvement, correct me if I'm wrong).

WDYT?

Reply:
Ah ok, I see what you're saying but the only thing to highlight is that the probing happens with range requests so it only has to pull down small amounts of the file, so for direct uploads with this code we're now copying the whole file to disk which would make performance worse. If it's a direct upload and there's no encryption we should avoid the copy to disk here I think.

Reply:
I see, that makes sense. What do you think should be the right approach here then? Maybe deal with the encryption as part of the "copy to s3" part as streaming decryption? So the condition for copying to S3 becomes "is not a direct upload or has encryption". It would only affect (more copies) in the case of "direct upload + encryption", in which the file is already on s3 and we're streaming it back and up only for decryption. This shouldn't be too bad tho since I think it will be the rarest kind of input (most of the times it will be encrypted in some dStorage).

Reply:
> I see, that makes sense. What do you think should be the right approach here then? Maybe deal with the encryption as part of the "copy to s3" part as streaming decryption? So the condition for copying to S3 becomes "is not a direct upload or has encryption". It would only affect (more copies) in the case of "direct upload + encryption", in which the file is already on s3 and we're streaming it back and up only for decryption. This shouldn't be too bad tho since I think it will be the rarest kind of input (most of the times it will be encrypted in some dStorage).

Yeah that sounds right to me 👍